lib/rbs/unit_test/with_aliases.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+module RBS
+  module UnitTest
+    module WithAliases
+      include Convertibles
+
+      class WithEnum
+        include Enumerable
+
+        def initialize(enum) = @enum = enum
+
+        def each(&block) = @enum.each(&block)
+
+        def and_nil(&block)
+          self.and(nil, &_ = block)
+        end
+
+        def but(*cases, &block)
+          return WithEnum.new to_enum(__method__ || raise, *cases) unless block
+
+          each do |arg|
+            yield arg unless cases.any? { (_ = _1) === arg }
+          end
+        end
+
+        def and(*args, &block)
+          return WithEnum.new to_enum(__method__ || raise, *args) unless block
+
+          each(&block)
+          args.each do |arg|
+            if WithEnum === arg # use `===` as `arg` might not have `.is_a?` on it
+              arg.each(&block)
+            else
+              block.call(_ = arg)
+            end
+          end
+        end
+      end
+
+      def with(*args, &block)
+        return WithEnum.new to_enum(__method__ || raise, *args) unless block
+        args.each(&block)
+      end
+
+      def with_int(value = 3, &block)
+        return WithEnum.new to_enum(__method__ || raise, value) unless block
+        yield value
+        yield ToInt.new(value)
+      end
+
+      def with_float(value = 0.1)
+        return WithEnum.new to_enum(__method__ || raise, value) unless block_given?
+        yield value
+        yield ToF.new(value)
+      end
+
+      def with_string(value = '')
+        return WithEnum.new to_enum(__method__ || raise, value) unless block_given?
+        yield value
+        yield ToStr.new(value)
+      end
+
+      def with_array(*elements)
+        return WithEnum.new to_enum(__method__ || raise, *elements) unless block_given?
+
+        yield elements
+        yield ToArray.new(*elements)
+      end
+
+      def with_hash(hash = {})
```

Comment:
`with_array(*elements)`, but not `with_hash(**elements)`?

Note: kwargs isn’t Symbols only and that – good?
```ruby
p('any key' => 'is valid', 420 => 69)
  #=> {"any key"=>"is valid", 420=>69}
```

Reply:
We can pass hash without curly brace with the current definition. So any other good reason to make it kwargs?

Reply:
I just prefer consistency.